secure_message/services/party_service.py

```diff
@@ -10,18 +10,30 @@
 
 
 class PartyService:
+
+    users_cache = dict()
+    business_details_cache = dict()
+
     @staticmethod
-    def get_business_details(ru):
-        """Retrieves the business details from the party service"""
+    def get_url(api_param, code):
+        """
+        :param api_param:  is the key configuration that that represent part or the URI .
+        :param code: is the code to use in the url ( uuid or ru )
+        :return: a formatted url
+        """
+        if current_app.config[api_param] is None or current_app.config['RAS_PARTY_SERVICE'] is None:
+            raise KeyError("%s not present" % api_param)
+        return current_app.config[api_param].format(current_app.config['RAS_PARTY_SERVICE'], code)
 
-        url = current_app.config['RAS_PARTY_GET_BY_BUSINESS'].format(current_app.config['RAS_PARTY_SERVICE'], ru)
-        party_data = requests.get(url, auth=current_app.config['BASIC_AUTH'], verify=False)
+    def get_business_details(self, ru):
+        """Retrieves the business details from the party service"""
 
-        logger.debug('Party service get business details result',
-                     status_code=party_data.status_code,
-                     reason=party_data.reason,
-                     text=party_data.text,
-                     url=url)
+        if ru not in self.business_details_cache:
+            party_data = requests.get(PartyService.get_url('RAS_PARTY_GET_BY_BUSINESS', ru),
+                                      auth=current_app.config['BASIC_AUTH'], verify=False)
+            self.users_cache.update(ru, party_data)
```

Comment:
Would expect to be using the business details cache Inside the `get_business_details` method: `users_cache` -> `business_details_cache`. (Same again a couple of lines below).